src/boot/ostree-prepare-root.service

```diff
@@ -17,7 +17,6 @@
 Description=OSTree Prepare OS/
 Documentation=man:ostree(1)
 DefaultDependencies=no
-ConditionKernelCommandLine=ostree
```

Comment:
This one is tricky.  What we wanted to defend against here is anyone who has the ostree dracut module installed but isn't booting ostree.

This is really the fundamental change; you're changing it here so that if we detect an ostree setup, we use it by default.  We may need to support something like `ostree=no` to disable it for oddball cases?

I'm not totally sure.

Maybe this isn't a real case anymore - quite a while ago we changed the ostree dracut module to be optionally opt-in (yes this is confusing).  See https://github.com/ostreedev/ostree/blob/c9271423b1c780ccb9ff47685d22d744968d9d19/configure.ac#L504
which is used by Fedora derivatives today.  This means we don't inject the module by default, but require invokers of dracut to use e.g. `--add ostree` explicitly, and this is done by rpm-ostree for example.

OK so, perhaps this is just OK to land.

Reply:
Our oddball case for the record is we are gonna read some GPT flag to see if we want to boot the latest symlink or rollback once  (that's all that we require)... But I haven't got that far yet, so at the moment, it's just boot the latest symlink!

Reply:
`ConditionKernelCommandLine=ostree` is present in a lot of units outside of this repo as well. For alignment, it's probably cleaner to just leave it there.

One approach is to have something like `ostree=latest`. Then `ConditionKernelCommandLine=ostree` is still valid, but the kargs don't need to change on each update. It also makes the logic to boot the latest symlink explicit.

Reply:
ostree=latest works for me!

Reply:
A question for both of you @cgwalters @jlebon . The BLS files exist, but are not readable by the bootloader. But we can read them just fine in the initramfs, by changing this code up a little...

I was wondering what's the most bulletproof way of identifying the most recent BLS .conf file and the second most recent BLS .conf? Then we can just take the ostree= from there, I think this could be a simple solution to formulate the full path. (this might make more sense looking at the output below, BLS and /proc/cmdline are not consistent because it doesn't really matter when your bootloader cannot parse BLS)

@alexlarsson was also suggesting earlier, it could be possible that maybe we always make ostree-0-centos.conf the latest one, then we can guarantee ostree-1-centos.conf is the rollback one (or vice-versa if it's easier).

Could this work, without breaking anything existing in ostree? Still learning all the little nuances of ostree day by day :smile:

Reply:
The guaranteeing ostree-0-centos.conf and ostree-1-centos.conf are latest and the second most latest is kinda nice as you don't have to worry about timestamps getting messed up somehow.

Reply:
I do still think though that we can add some sort of flag to the initramfs that says "require ostree"; maybe `/usr/lib/ostree-required` or whatever.

Then units in the initramfs can use `ConditionPathExists` on that.

Or, any units that should run in the initramfs iff ostree is enabled could be pulled in via `WantedBy=ostree-prepare-root.service`, though that gets a bit trickier to manage.

(But this is a distinct thread from the suggestion of having code in the initramfs parse the BLS configs)